packages/smithy-aws-core/src/smithy_aws_core/endpoints/standard_regional.py

```diff
@@ -0,0 +1,61 @@
+#  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#  SPDX-License-Identifier: Apache-2.0
+from dataclasses import dataclass
+from urllib.parse import urlparse
+
+import smithy_core
+from smithy_core import URI
+from smithy_http.aio.interfaces import EndpointResolver
+from smithy_http.endpoints import Endpoint, EndpointResolutionError
+
+
+@dataclass(kw_only=True)
+class RegionalEndpointParameters:
+    """Endpoint parameters for services with standard regional endpoints."""
+
+    sdk_endpoint: str | smithy_core.interfaces.URI | None
+    region: str | None
+
+
+class StandardRegionalEndpointsResolver(EndpointResolver[RegionalEndpointParameters]):
+    """Resolves endpoints for services with standard regional endpoints."""
+
+    def __init__(self, endpoint_prefix: str):
+        self._endpoint_prefix = endpoint_prefix
+
+    async def resolve_endpoint(self, params: RegionalEndpointParameters) -> Endpoint:
+        if params.sdk_endpoint is not None:
+            # If it's not a string, it's already a parsed URI so just pass it along.
+            if not isinstance(params.sdk_endpoint, str):
+                return Endpoint(uri=params.sdk_endpoint)
+
+            # Does crt have implementations of these parsing methods? Using the standard
+            # library is probably fine.
+            parsed = urlparse(params.sdk_endpoint)
```

Comment:
The CRT doesn't have any way to parse it in their public bindings to my knowledge. This should be fine for now.